src/components/user.js

```diff
@@ -15,69 +26,143 @@ export default class User extends React.Component {
     }
 
-    handleClickAddress (e) {
+    _handleClickAddress (e) {
         this.setState({ showAddress: !this.state.showAddress });
         e.preventDefault();
     }
 
-    handleClickCompany (e) {
+    _handleClickCompany (e) {
         this.setState({ showCompany: !this.state.showCompany });
         e.preventDefault();
     }
 
-    handleClickEdit (e) {
+    _handleClickEdit (e) {
         const { currentUser, wantEdit } = this.state;
 
         if (wantEdit) {
             const newUser = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
 
             UserAction.updateUser(newUser);
         }
-
         this.setState({ wantEdit: !wantEdit });
         e.preventDefault();
     }
 
-    handleChange (e, type) {
-        const { currentUser, wantEdit } = this.state;
-        const {isNewUser} = this.props;
+    _handleChange (e, obj) {
+        const { currentUser, wantEdit, formErrors } = this.state;
+        const { isNewUser }                         = this.props;
 
         if (wantEdit || isNewUser) {
-            currentUser[type][e.target.name] = e.target.value;
-            this.setState({ currentUser });
+            const name  = e.target.name;
+            const value = e.target.value;
+
+            currentUser[obj][name] = value;
+            formErrors[name]       = this._validateField(name, value);
+            this.setState({ currentUser, formErrors });
+        }
+        e.preventDefault();
+    }
+
+    _addingFailed = userId => {
+        const { isNewUser }                = this.props;
+        const { formErrors, addDuplicate } = this.state;
+
+        if (isNewUser && userId > 0) {
+            formErrors.duplicateUser.text   = ERRORS.userExist;
+            formErrors.duplicateUser.userId = `#${userId}`;
+            this.setState({ formErrors: formErrors, addDuplicate: !addDuplicate });
+        }
+    };
+
+    _handleClickSubmit (e) {
+        const { currentUser, addDuplicate, formErrors } = this.state;
+        const newUser                                   = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
+        const resultValid                               = this._validateFields(currentUser.info);
+
+        if (resultValid.nameValid === '' && resultValid.phoneValid === '' && resultValid.emailValid === '') {
+            UserAction.addNewUser(newUser, addDuplicate);
+        }
+        else {
+            formErrors.name  = resultValid.nameValid;
+            formErrors.phone = resultValid.phoneValid;
+            formErrors.email = resultValid.emailValid;
+            this.setState({ formErrors });
         }
         e.preventDefault();
     }
 
-    handleClickSubmit () {
-        const { currentUser } = this.state;
-        const newUser         = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
+    componentDidMount () {
+        UserStore.on(EVENT_TYPE.change, this.props.onChange);
```

Comment:
I still can see warning in the console about too may listeners.
To reproduce open the application page and click on `Add new user` button.

See: we pass `onChange` handler from `user-list` to **each** `user` just to subscribe for `change` event and use this passed handler.
We have to subscribe in `user-list` for `change` event. And we've already done it (see [here](https://github.com/viktoria2506/user-list/blob/master/src/components/user-list.js#L35)).
So we can **remove**  subscription for `EVENT_TYPE.change` from the `user` and restore it in the `user-list`.

But subscriptions for `EVENT_TYPE.addingFailed` and `EVENT_TYPE.userAdded` we have to move to the `user-list`